### PR TITLE
BlockProcessor interface change

### PR DESCRIPTION
**Pull request by sasurobert**
## Description of the reasoning behind the pull request (what feature was missing / how the problem was manifesting itself / what was the motive behind the refactoring)
- 
- 
- 
  
## Proposed Changes
- 
- 
- 

## Testing procedure
- 
- 
-

### CODE REVIEW

File: update/mock/transactionCoordinatorMock.go [hunk 1]
```diff
@@ -117,12 +117,12 @@ func (tcm *TransactionCoordinatorMock) RemoveTxsFromPool(body *block.Body) error
...
 // ProcessBlockTransaction -
-func (tcm *TransactionCoordinatorMock) ProcessBlockTransaction(header data.HeaderHandler, body *block.Body, haveTime func() time.Duration) error {
+func (tcm *TransactionCoordinatorMock) ProcessBlockTransaction(header data.HeaderHandler, body *block.Body, haveTime func() time.Duration) (block.MiniBlockSlice, error) {
 	if tcm.ProcessBlockTransactionCalled == nil {
-		return nil
+		return nil, nil
 	}
 
-	return tcm.ProcessBlockTransactionCalled(header, body, haveTime)
+	return nil, tcm.ProcessBlockTransactionCalled(header, body, haveTime)
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply update/mock/transactionCoordinatorMock.go:125]: done

File: process/mock/transactionCoordinatorMock.go [hunk 2]
```diff
@@ -126,12 +126,12 @@ func (tcm *TransactionCoordinatorMock) RemoveTxsFromPool(body *block.Body) error
...
 // ProcessBlockTransaction -
-func (tcm *TransactionCoordinatorMock) ProcessBlockTransaction(header data.HeaderHandler, body *block.Body, haveTime func() time.Duration) error {
+func (tcm *TransactionCoordinatorMock) ProcessBlockTransaction(header data.HeaderHandler, body *block.Body, haveTime func() time.Duration) (block.MiniBlockSlice, error) {
 	if tcm.ProcessBlockTransactionCalled == nil {
-		return nil
+		return nil, nil
 	}
 
-	return tcm.ProcessBlockTransactionCalled(header, body, haveTime)
+	return nil, tcm.ProcessBlockTransactionCalled(header, body, haveTime)
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply process/mock/transactionCoordinatorMock.go:134]: done

File: process/mock/preprocessorMock.go [hunk 3]
```diff
@@ -77,11 +77,11 @@ func (ppm *PreProcessorMock) SaveTxsToStorage(body *block.Body) error {
...
 
 // ProcessBlockTransactions -
-func (ppm *PreProcessorMock) ProcessBlockTransactions(header data.HeaderHandler, body *block.Body, haveTime func() bool) error {
+func (ppm *PreProcessorMock) ProcessBlockTransactions(header data.HeaderHandler, body *block.Body, haveTime func() bool) (block.MiniBlockSlice, error) {
 	if ppm.ProcessBlockTransactionsCalled == nil {
-		return nil
+		return nil, nil
 	}
-	return ppm.ProcessBlockTransactionsCalled(header, body, haveTime)
+	return nil, ppm.ProcessBlockTransactionsCalled(header, body, haveTime)
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply process/mock/preprocessorMock.go:84]: done

File: integrationTests/mock/transactionCoordinatorMock.go [hunk 4]
```diff
@@ -126,12 +126,12 @@ func (tcm *TransactionCoordinatorMock) RemoveTxsFromPool(body *block.Body) error
...
 // ProcessBlockTransaction -
-func (tcm *TransactionCoordinatorMock) ProcessBlockTransaction(header data.HeaderHandler, body *block.Body, haveTime func() time.Duration) error {
+func (tcm *TransactionCoordinatorMock) ProcessBlockTransaction(header data.HeaderHandler, body *block.Body, haveTime func() time.Duration) (block.MiniBlockSlice, error) {
 	if tcm.ProcessBlockTransactionCalled == nil {
-		return nil
+		return nil, nil
 	}
 
-	return tcm.ProcessBlockTransactionCalled(header, body, haveTime)
+	return nil, tcm.ProcessBlockTransactionCalled(header, body, haveTime)
```
[the remark this answers is not in the document]
Comment (sasurobert) [reply integrationTests/mock/transactionCoordinatorMock.go:134]: done

**Review by sstanculeanu — APPROVED** [review]

**Review by sstanculeanu — APPROVED** [review]

**Review by SebastianMarian — APPROVED** [review]

**Pull request merged**